woocommerce/wc-admin
# Remove categories without menu items

Fixes #6314

* Removes categories without menu items
* Removes capabilities on categories and leaves responsibility to menu items.  This one is up for discussion, but my thinking here is that this could be confusing to have an item a user could access, but they cannot see due to category restrictions.

### Screenshots

<img width="633" alt="Screen Shot 2021-02-11 at 5 20 44 PM" src="https://user-images.githubusercontent.com/10561050/107706298-147fae80-6c8e-11eb-8cf5-9b0f378815cd.png">


### Detailed test instructions:

1. Load up a dev branch of Gutenberg and activate (needed for new `hideIfTargetMenuEmpty` prop).
2. Create a user and role with capabilities that can see some items, but not all.
3. `return true;` if you have trouble seeing the nav in `user_can_analytics()`.  Issue - https://github.com/woocommerce/woocommerce-admin/issues/6315
4. Note that categories without items are hidden.
